### DISCUSSION TITLE
Save address

### DISCUSSION THREAD

**Discussion by lalo-uy** [arendst/Tasmota#22040] [Q&A]
On status 0 of a sonoff basic y see:
"SaveCount":15748,"SaveAddress":"F8000"

Is ther a way to change SaveAddress? May be by recompiling. On witch source file is that set?

**Comment by sfromis** [comment]
What do you want to solve? It will normally be changed every time you save, cycling in 8 slots. `SetOption12` exists to disable the rotation, but not recommended.

**Comment by lalo-uy** [comment]
Ok, 8 slots will be ok.
I was trying to extend the life of that devices by flashing a different
addr.
There was a failure in the central control and the heater was switched very
quickly for 2 days. :-(

**Reply by sfromis** [reply]
My preference is to normally run with `SetOption0 0` to disable automatic save of power state, and a default power state (typically off) using `PowerOnState`, as I don't want an old state to kick in after a power outage. I'm not seeing it as even desirable to have power state saved across outages of whatever length.

(Of course `PowerOnState 4` for always-on)

**Comment by reimerp** [comment]
I also use `SetOption0 0` and `ProwerOnState 0`but I have also three devices that have very high SaveCount Counter. (~17k) . These devices also have a similar high BootCount, so I checked: On every boot a save is done.

Can this be prevented completely?

I assume these devices do a triggered reboot, either on bad WLAN or dropped MQTT connection, since they have the longest distance to my router

**Reply by sfromis** [reply]
Well, you should not get that many reboots, unless you already know why it is happening. What happens if your Wifi is "too horrible" may be up to your configuration, including `WifiConfig`.

Anyway, another config to control saving is the command `SaveData`.
